### PR TITLE
feat: added Videos  and categories context

### PR DESCRIPTION
**Pull request by Chakravarthi0**
added Videos  and categories context

### CODE REVIEW

File: src/context/CategoryContext.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        const response = await axios.get("/api/categories");
+
+        if (response.status === 200) {
+          setCategories(response.data.categories);
+        }
+      } catch (err) {
+        console.log(err);
+      }
+    })();
+  }, []);
```
Comment (poornima-alugubelly) [inline src/context/CategoryContext.jsx:23]: there is no need to give empty array dependency here as context will run only once anyways

File: src/context/VideosContext.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          setVideos(response.data.videos);
+        }
+      } catch (err) {
+        console.log(err);
+        setError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    })();
+  }, []);
```
Comment (poornima-alugubelly) [inline src/context/VideosContext.jsx:29]: same here

File: src/pages/home/Home.jsx [hunk 3]
```diff
@@ -1,21 +1,37 @@
...
+        <div className="videos-container">
+          {videos.map((video) => (
+            <VideoCard key={video._id} video={video} />
+          ))}
+          {videos.map((video) => (
+            <VideoCard key={video._id} video={video} />
+          ))}
+          {videos.map((video) => (
+            <VideoCard key={video._id} video={video} />
+          ))}
```
Comment (poornima-alugubelly) [inline src/pages/home/Home.jsx:32]: is there any reason for making three mappings here? if not you can make a single map

**Review by poornima-alugubelly — APPROVED** [review]

**Review by msdeshmukh009 — APPROVED** [review]
The code looks clean. 
The UI is looking very good.
Good to merge.

**Pull request merged**